### ISSUE TITLE
[fvt]2.13.3 rmvm -f -p on the vm using  "vmstorage=phy:/ " output libvirt error but the command could not send any errors

### ISSUE THREAD

**Issue by junxiawang** [xcat2/xcat-core#2562]
env:ppc64le
build:
  c910f03c17k27:~ # lsxcatd -v
Version 2.13.2 (git commit dab564dd7f23cb8bc46119f11d235de33c57802a, built Wed Feb 22 22:42:07 EST 2017)
c910f03c17k27:~ # 
How to reproduce:
```
1.lsdef c910f03c17k28
Object name: c910f03c17k28
    arch=ppc64le
    cons=kvm
    currchain=boot
...
    vmmemory=6144
    vmnicnicmodel=virtio
    vmnics=brenP3p9s0f0
    vmstorage=phy:/dev/mapper/vdiskvg03-vdisk01n28

2.rmvm c910f03c17k28 -f -p
echo $?
0
3 In fact in our log:
Feb 27 02:21:55 c910f03c17k27 xcat[8069]:  xCAT: Allowing rmvm to c910f03c17k29 -f -p for root from localhost
Feb 27 02:21:56 c910f03c17k27 xcat[8072]:  ERR kvm: libvirt error code: 50, message: Storage volume not found: no storage vol with matching path ''

```

**Comment by whowutwut** [comment]
@gurevichmark are you familiar with this?

**Comment by gurevichmark** [comment]
@junxiawang Am trying to recreate this problem on my system.
How do you setup the storage on the hypervisor so that a VM can specify `phy:` ?

On my `fs2` hypervisor, the `/dev/mapper` looks like this:

```
[root@fs2 ~]# ls -l /dev/mapper
total 0
crw-------. 1 root root 10, 236 Jan  4 15:11 control
lrwxrwxrwx. 1 root root       7 Jan  4 15:11 ibmpkvm_rootvg-ibmpkvm_lv_data -> ../dm-3
lrwxrwxrwx. 1 root root       7 Jan  4 15:11 ibmpkvm_rootvg-ibmpkvm_lv_swap -> ../dm-0
lrwxrwxrwx. 1 root root       7 Jan  4 15:11 ibmpkvm_rootvg-ibmpkvm_lv_system -> ../dm-2
lrwxrwxrwx. 1 root root       7 Jan  4 15:11 ibmpkvm_rootvg-ibmpkvm_lv_var -> ../dm-1
[root@fs2 ~]#
```

**Comment by junxiawang** [comment]
1. I set up vm using mkvm 
2.from your fs2 I get that you define the vm using "vmstorage=dir:///var/lib/libvirt/images",mine using "vmstorage=phy:/dev/mapper/vdiskvg03-vdisk01n28"

**Comment by gurevichmark** [comment]
@junxiawang But how did you choose `/dev/mapper/vdiskvg03-vdisk01n28` ? 

Did you have something configured on the hypervisor **before** running `mkvm` that created `vdiskvg03-vdisk01n28` under `/dev/mapper` ?
OR
I can specify something like `vmstorage=phy:/dev/mapper/abc` and the directory `abc` will be created for me when I run `mkvm` ?

**Comment by gurevichmark** [comment]
Fixed by PR #2631 . A warning message will be logged when it is not appropriate to remove storage which was not created as part of VM creation.

**Comment by junxiawang** [comment]
OK I will retest the issue
